internal/gbimporter/gbimporter.go

```diff
@@ -63,6 +68,11 @@ func (i *importer) ImportFrom(path, srcDir string, mode types.ImportMode) (*type
 	buildDefaultLock.Lock()
 	defer buildDefaultLock.Unlock()
 
+	// return the package if it's in the cache
+	if i.cache[path] != nil {
```

Comment:
the old gocode had a check on the last modification date of the file to avoid unnecessary work:
https://github.com/nsf/gocode/blob/7b1d4e18cdc58a74dc1bd4c2d45b3f1b2ca227c3/declcache.go#L75-L91

Have you checked if something like this can be implemented here too?

Reply:
The old gocode parsed a compiled pkg file.  This doesn't know about the compiled package file to do a stat on.